src/UnoDrive.Shared/ViewModels/MyFilesViewModel.cs

```diff
@@ -142,7 +149,7 @@ async Task LoadDataAsync(string pathId = null, Action callback = null)
 				await currentLoadDataTask.Task;
 			}
 
-			currentLoadDataTask = new TaskCompletionSource<bool>();
+			currentLoadDataTask = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
This is a good thing to do when you create a `TaskCompletionSource`, as you [can read here](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#always-create-taskcompletionsourcet-with-taskcreationoptionsruncontinuationsasynchronously)

Reply:
That is a very useful document for TCS, thanks for sharing!